### PR TITLE
[YUNIKORN-2655] Cleanup REST API documentation

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
The REST API documentation is not up to date with the current behaviour as it does not show any 400 or 404 errors returned by a number of API calls.

The error response only shows a 500 code with the same message for each call.

We should move to a simple list for each call showing the applicable errors like this:
```
### Error responses

**Code** : `400 Bad Request` (URL query is invalid, missing partition name)

**Code** : `404 Not Found` (Partition not found)

**Code** : `500 Internal Server Error` 
```
Remove the error examples as they do not add any detail required

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [x] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2655

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: docs/api/scheduler.md [hunk 1]
```diff
@@ -1705,6 +1621,9 @@ yunikorn_scheduler_vcore_nodes_usage{range="(80%,90%]"} 0
 yunikorn_scheduler_vcore_nodes_usage{range="(90%,100%]"} 0
 yunikorn_scheduler_vcore_nodes_usage{range="[0,10%]"} 0
 ```
+### Error response
+
+**Code** : `500 Internal Server Error`
```
Comment (chenyulin0719) [inline docs/api/scheduler.md:1626]: I didn't find the code for '/ws/v1/metrics' to return 500. Is this a misplace?

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
@ryankert01 Thanks for the carefully check.  I found a mistake and also have a question.

File: docs/api/scheduler.md [hunk 2]
```diff
@@ -575,17 +569,12 @@ In the example below there are three allocations belonging to two applications,
 
 ### Error response
 
-**Code** : `500 Internal Server Error`
+**Code** : `400 Bad Request` (URL query is invalid)
 
-**Content examples**
+**Code** : `404 Not Found` (Partition or Application not found)
```
[the remark this answers is not in the document]
Comment (ryankert01) [reply docs/api/scheduler.md]: Done.

File: docs/api/scheduler.md (hunk 1 above)
Comment (ryankert01) [reply docs/api/scheduler.md]: Yep, it's a mistake. Thank you for pointed that out!

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**